### PR TITLE
Add purger to ExecutingStatementResource

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 1]
```diff
@@ -106,6 +114,35 @@ public ExecutingStatementResource(
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+
+        queryPurger.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        // snapshot the queries before checking states to avoid registration race
+                        for (Entry<QueryId, Query> entry : ImmutableSet.copyOf(queries.entrySet())) {
```
Comment (martint) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]: There's no race here. You can just iterate over the query set directly.

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 2]
```diff
@@ -106,6 +113,34 @@ public ExecutingStatementResource(
...
+                                queries.remove(entry.getKey());
+                            }
+                        }
+                    }
+                    catch (Throwable e) {
+                        log.warn(e, "Error removing old queries");
+                    }
+                },
+                200,
+                200,
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]: Does it have to be frequent?

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 3]
```diff
@@ -106,6 +113,34 @@ public ExecutingStatementResource(
...
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+
+        queryPurger.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        for (Entry<QueryId, Query> entry : queries.entrySet()) {
+                            // forget about this query if the query manager is no longer tracking it
+                            try {
+                                queryManager.getQueryInfo(entry.getKey());
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]: `#getQueryInfo` isn't free to call. Maybe `#getQueryState` would be cheaper and with the same effect of checking the query id validity? (or a dedicated method)

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java (hunk 2 above)
Comment (dain) [reply presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]: This is a copy of the purger from `QueuedStatementResource`... this basically restores the old behavior

**Pull request closed without merging**